File: 2022-2023/sourcecode/lab/lab13/OfflineCachingBasicsJava/app/src/main/java/pl/udu/uwr/pum/offlinecachingbasicsjava/data/converters/SubscriptionConverter.java
package pl.udu.uwr.pum.offlinecachingbasicsjava.data.converters;

import androidx.room.TypeConverter;

import com.google.gson.Gson;

import pl.udu.uwr.pum.offlinecachingbasicsjava.data.response.Subscription;

public class SubscriptionConverter {
    @TypeConverter
    public static String fromSubscription(Subscription subscription){
        Gson gson = new Gson();
        return gson.toJson(subscription);
    }

    @TypeConverter
    public  static Subscription toSubscription(String subscription){
        Gson gson = new Gson();
        return gson.fromJson(subscription, Subscription.class);
    }
}

File: 2022-2023/sourcecode/lab/lab13/OfflineCachingBasicsKotlin/app/src/main/java/pl/udu/uwr/pum/offlinecachingbasicskotlin/data/converters/SubscriptionConverter.kt
package pl.udu.uwr.pum.offlinecachingbasicskotlin.data.converters

import androidx.room.TypeConverter
import pl.udu.uwr.pum.offlinecachingbasicskotlin.data.reponse.Subscription

object SubscriptionConverter {

    @TypeConverter
    @JvmStatic
    fun fromSubscription (subscription: Subscription): String{
        return subscription.toString()
    }

    @TypeConverter
    @JvmStatic
    fun toSubscription(subscription: String): Subscription {
        val payment_method = subscription.substringAfter("payment_method=").substringBefore(",")
        val plan = subscription.substringAfter("plan=").substringBefore(",")
        val status = subscription.substringAfter("status=").substringBefore(",")
        val term = subscription.substringAfter("term=").substringBefore(")")

        return Subscription(payment_method, plan, status, term)
    }
}